server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/Enqueuer.java

```diff
@@ -76,6 +76,14 @@ void enQueue(Mail mail) throws MailQueue.MailQueueException {
             .block();
     }
 
+    Mono<Void> reQueue(Mail mail) throws MailQueue.MailQueueException {
+        EnqueueId enqueueId = EnqueueId.generate();
+        return saveMail(mail)
+            .map(partIds -> new MailReference(enqueueId, mail, partIds))
+            .flatMap(Throwing.function(this::publishReferenceToRabbit).sneakyThrow())
+            .thenEmpty(Mono.fromRunnable(enqueueMetric::increment));
```

Comment:
We technicaly do not have more enqueued emails. I would expect requeue to be a noop operation for this metric.